Fixes issue #126

NiteshKant opened the pull request:
HttpServer now handles HTTP 1.0 requests as per the spec.

Commit:
Fixes issue #126

HttpServer now handles HTTP 1.0 requests as per the spec.

cloudbees-pull-request-builder commented:
[RxNetty-pull-requests #52](https://netflixoss.ci.cloudbees.com/job/RxNetty-pull-requests/52/) ABORTED

Commit:
Fix for issue #126 was wrong.

HttpServerResponse.close() was asserting the transfer encoding to add LastHttpContent.
The transfer encoding was getting added in ServerRequestResponseConverter.
In cases, when close() is NOT invoked from the eventloop of the server, there is a race-condition
between when the transfer encoding is updated (in the handler) and when close() asserts the same as
the handler is invoked in a different thread (server eventloop) than close()

Also added some testcases for this scneario.

Bug fix

cloudbees-pull-request-builder commented:
[RxNetty-pull-requests #53](https://netflixoss.ci.cloudbees.com/job/RxNetty-pull-requests/53/) FAILURE
Looks like there's a problem with this pull request

NiteshKant commented:
> > > RxNetty-pull-requests #53 FAILURE
> > > Looks like there's a problem with this pull request

Strange that the build failed due to maven not reachable.

```
* What went wrong:
Could not resolve all dependencies for configuration ':rx-netty:testCompile'.
> Could not resolve junit:junit:4.10.
  Required by:
      com.netflix.rxnetty:rx-netty:0.3.6
   > Could not GET 'http://repo1.maven.org/maven2/junit/junit/4.10/junit-4.10.pom'.
      > No route to host
```

Commit:
Making the rfc 2145 compliance optional.

http://tools.ietf.org/html/rfc2145#section-2.3 says that a HTTP 1.1 compliant server should always send a 1.1 response.
However, some clients (apache bench) can not gracefully handle 1.1 responses. Thus making this optional on the HTTPServerBuilder.

cloudbees-pull-request-builder commented:
[RxNetty-pull-requests #54](https://netflixoss.ci.cloudbees.com/job/RxNetty-pull-requests/54/) SUCCESS
This pull request looks good
